BEIMA.Client/src/pages/Devices/AddDevicePage.js

```diff
@@ -0,0 +1,98 @@
+import { useOutletContext } from 'react-router-dom';
+import {  Card, Button, Dropdown, Row, Col, Form } from 'react-bootstrap';
+import { useEffect, useState } from "react";
+import styles from './AddDevicePage.module.css';
+import FormList from '../../shared/FormList/FormList.js';
+import ImageFileUpload from '../../shared/ImageFileUpload/ImageFileUpload.js';
+
+const AddDevicePage = () => {
+  // this will be replaced with API call based on selected device type to get the fields
+  const defaultDeviceFields = {
+    "Building": "",
+    "Longitude": "",
+    "Latitude": "",
+    "Device Tag": "",
+    "Manufacturer": "",
+    "Model Number": "",
+    "Serial Number": "",
+    "Year Manufactured": "",
+    "Notes": ""
+  }
+
+  const [deviceFields, setDeviceFields] = useState(defaultDeviceFields);
+  const [setPageName] = useOutletContext();
+  const [deviceImage, setDeviceImage] = useState();
+  const [deviceAdditionalDocs, setAdditionalDocs] = useState();
+  const [fullDeviceJSON, setFullDeviceJSON] = useState({});
+
+  useEffect(() => {
+    setPageName('Add Device')
+  })
+
+  // gathers all the input and puts it into JSON, files are just assigned to state variables for now
+  function createJSON(event){
+    let formFields = event.target.form.elements;
+    let fieldValues = {};
+
+    for(let i = 0; i < formFields.length; i++){
+      let formName = formFields[i].name;
+      let fieldNames = Object.keys(deviceFields);
+
+      if(fieldNames.includes(formName)){
+        let formJSON =  {[formName] : formFields[i].value};
+        formFields[i].value = "";
+        Object.assign(fieldValues, formJSON);
+      } else if (formName === "Device Image"){
+        setDeviceImage(formFields[i].files[0]);
+        formFields[i].value = "";
+      } else if (formName === "Additional Documents"){
+        setAdditionalDocs(formFields[i].files);
+        formFields[i].value = "";
+      }
+    }
+
+    setDeviceFields(defaultDeviceFields);
+    setFullDeviceJSON(fieldValues);
+    console.log(fullDeviceJSON);
+    console.log(deviceImage);
+    console.log(deviceAdditionalDocs);
```

Comment:
Leftover from debugging?

Reply:
Yep, got rid of console statements

Reply:
Actually, I need them there. They are printing the created JSONs and files uploaded, and besides printing them, they are not used and that causes the client build to fail. When we add the API calls to the pages that won't be a problem anymore